### DISCUSSION TITLE
Combining meshes

### DISCUSSION THREAD

**Discussion by sinaomrani96** [idaholab/moose#25430] [Q&A Meshing]
Hi,
Hope you are well.

I was wondering if there is a way to combine two meshes in which we can keep the blocks in one of them and elements from other one. Assuming that both of them are identical. Thanks.

**Comment by GiudGiud** [comment]
Hello

so you want only the elements from one of the meshes? But use the subdomain names from the other?

i feel like using the combinerGenerator, the renameSubdomainGenerator and the blockDeletionGenerator (once to remove and once to add elements to a domain) you can make that work.

**Reply by sinaomrani96** [reply]
Thank you @GiudGiud, that gives me some ideas. On another note, is there a way to do adaptivity on a mesh that refines around blocks interfaces and coarsen anywhere else? I am trying this and only get the refinement around the subdomains but the coarsening on the domain do not work.

**Reply by GiudGiud** [reply]
We can refine on interfaces using boundary markers for example.

we can only coarsen what has been previously refined. So if you generate a very coarse mesh you can refine it once at the beginning then coarsen again for regions that don’t need the refinement

**Reply by sinaomrani96** [reply]
Hi @GiudGiud,

As you mentioned we only can coarsen what has been previously refined, I was wondering if we can remesh a mesh. The issue is that I have an image and to capture its features I need to match the number of pixels with the elements using `GeneratedMeshGenerator` and `ImageSubdomainGenerator`. However, this leads to computationally expensive problem and I want to use adaptivity to refine only around blocks and boundaries. I followed your suggestion of using `blockDeletionGenerator` but I need to remesh it first and then refine the boundaries. I was wondering if you have any suggestions. Thanks.

**Reply by GiudGiud** [reply]
Hello

if you delete an entire volume you can remesh that volume using the XYDelaunayMG in 2D

You can also probably use external software like cubit to import the mesh and remesh parts of it

Guillaume

**Reply by sinaomrani96** [reply]
Dear @GiudGiud,

Thanks for the help. Sorry to ask too many questions, but I was wondering if there is a way to control the minimum element size after adaptivity. I do not know the calculation behind it, and would like to know how I can control the minimum element size based on initial number of elements, initial_steps, and max_h_level etc. Thanks.

**Reply by GiudGiud** [reply]
You can bake that criterion inside a custom indicator or a custom marker
